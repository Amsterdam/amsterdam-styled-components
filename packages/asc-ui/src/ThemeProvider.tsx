import * as React from 'react'
import { StyledComponents, Theme } from '@datapunt/asc-core'
import defaultTheme from './config/theme'

interface Props {
  theme?: Theme.ThemeInterface,
  children: any
}

const ASCDefaultThemeProvider = ({ theme, children }: Props) => {
  const namespacedTheme = {
    ascTheme: Object.assign({}, defaultTheme.ascTheme, theme),
  }
  return (
    <StyledComponents.ThemeProvider theme={namespacedTheme}>
      {children}
    </StyledComponents.ThemeProvider>
  )
}

export default ASCDefaultThemeProvider
