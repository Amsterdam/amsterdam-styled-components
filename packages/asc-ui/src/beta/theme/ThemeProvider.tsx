import deepMerge from 'deepmerge'
import type { FunctionComponent, ReactNode } from 'react'
import { ThemeProvider as AscThemeProvider } from 'styled-components'
import { ascBetaTheme } from '.'
import type { Theme } from '../types'

export interface ThemeProviderProps {
  overrides?: Partial<Theme.ThemeInterface>
  theme?: Theme.ThemeInterface
  deep?: boolean
  children: ReactNode
}

const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  theme: defaultTheme = ascBetaTheme,
  overrides,
  deep = true,
  children,
}) => {
  const theme: Theme.ThemeInterface = deep
    ? deepMerge(defaultTheme, overrides || {})
    : { ...defaultTheme, ...overrides }

  return <AscThemeProvider theme={theme}>{children}</AscThemeProvider>
}

export default ThemeProvider
