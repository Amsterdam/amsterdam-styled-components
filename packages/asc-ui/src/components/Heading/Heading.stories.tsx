import React from 'react'
import Heading from './Heading'

export default {
  title: 'Experimental/Atoms/Typography/Heading',
}

export const DefaultStyle = () => (
  <>
    <Heading>Heading one</Heading>
    <Heading forwardedAs="h2">Heading two</Heading>
    <Heading forwardedAs="h3">Heading three</Heading>
    <Heading forwardedAs="h4">Heading four</Heading>
  </>
)
