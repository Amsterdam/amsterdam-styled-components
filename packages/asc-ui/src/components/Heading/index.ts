export { default } from './Heading'
export { default as HeadingStyle } from './HeadingStyle'
