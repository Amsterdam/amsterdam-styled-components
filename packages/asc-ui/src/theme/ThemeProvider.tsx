import React, { ReactNode } from 'react'
import deepMerge from 'deepmerge'
import { ThemeProvider as AscThemeProvider } from 'styled-components'
import { ascDefaultTheme } from '.'
import { Theme } from '../types'

interface Props {
  overrides?: Partial<Theme.ThemeInterface>
  theme?: Theme.ThemeInterface
  deep?: boolean
  children: ReactNode
}

const ThemeProvider: React.FC<Props> = ({
  theme: defaultTheme = ascDefaultTheme,
  overrides,
  deep = true,
  children,
}) => {
  const theme: Theme.ThemeInterface = deep
    ? deepMerge(defaultTheme, overrides || {})
    : { ...defaultTheme, ...overrides }

  return (
    <AscThemeProvider theme={theme}>
      <>{children}</>
    </AscThemeProvider>
  )
}

export default ThemeProvider
