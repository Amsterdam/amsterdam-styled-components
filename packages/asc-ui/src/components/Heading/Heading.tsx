import React from 'react'
import HeadingStyle from './HeadingStyle'

const Heading: React.FC<
  {
    as?: keyof JSX.IntrinsicElements | React.ComponentType<any>
  } & React.HTMLAttributes<HTMLHeadingElement>
> = ({ children, className, as, ...otherProps }) => (
  <HeadingStyle className={className} as={as} {...otherProps}>
    {children}
  </HeadingStyle>
)

Heading.defaultProps = {
  as: 'h1',
}

export default Heading
