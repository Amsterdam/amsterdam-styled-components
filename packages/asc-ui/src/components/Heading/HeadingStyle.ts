import styled, { css } from 'styled-components'
import type { Props as TypographyProps } from '../Typography/TypographyStyle'
import Typography from '../Typography'

export type Props = TypographyProps

export const HeaderStyleCSS = css`
  margin-top: 0;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`

export default styled(Typography)<Props>`
  ${HeaderStyleCSS}
`
