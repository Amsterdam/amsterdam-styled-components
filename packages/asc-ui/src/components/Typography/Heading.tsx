import React from 'react'
import { styledComponents } from '@datapunt/asc-core'
import HeadingStyle from './HeadingStyle'

const Heading: React.FC<styledComponents.StyledProps<any>> = ({
  children,
  ...otherProps
}) => <HeadingStyle {...otherProps}>{children}</HeadingStyle>

export default Heading
