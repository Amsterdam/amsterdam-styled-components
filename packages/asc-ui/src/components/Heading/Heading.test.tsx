import { render } from '@testing-library/react'
import Heading from './Heading'
import ThemeProvider from '../../theme/ThemeProvider'

describe('Heading', () => {
  it('should render', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <Heading data-testid="h1" as="h1">
          Foo
        </Heading>
        <Heading data-testid="h2" as="h2">
          Foo
        </Heading>
        <Heading data-testid="h3" as="h3">
          Foo
        </Heading>
        <Heading data-testid="h4" as="h4">
          Foo
        </Heading>
        <Heading data-testid="h5" as="h5">
          Foo
        </Heading>
        <Heading data-testid="h6" as="h6">
          Foo
        </Heading>
      </ThemeProvider>,
    )

    expect(getByTestId('h1')).toHaveStyleRule('font-size', '24px')
    expect(getByTestId('h2')).toHaveStyleRule('font-size', '20px')
    expect(getByTestId('h3')).toHaveStyleRule('font-size', '20px')
    expect(getByTestId('h4')).toHaveStyleRule('font-size', '18px')
    expect(getByTestId('h5')).toHaveStyleRule('font-size', '18px')
    expect(getByTestId('h6')).toHaveStyleRule('font-size', '16px')
  })
})
