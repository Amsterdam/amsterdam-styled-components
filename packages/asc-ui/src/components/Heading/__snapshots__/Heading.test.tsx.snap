// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render 1`] = `
Array [
  .c1 {
  margin: 0;
  color: #000000;
  font-weight: 700;
  font-size: 24px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 30px;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c0 {
  margin-top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media screen and (min-width:1024px) {
  .c1 {
    font-size: 30px;
    line-height: 38px;
  }
}

<h1
    class="c0 c1"
  >
    Foo
  </h1>,
  .c1 {
  margin: 0;
  color: #000000;
  font-weight: 700;
  font-size: 20px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 28px;
  margin-bottom: 8px;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c0 {
  margin-top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media screen and (min-width:1024px) {
  .c1 {
    font-size: 24px;
    line-height: 30px;
  }
}

<h2
    class="c0 c1"
  >
    Foo
  </h2>,
  .c1 {
  margin: 0;
  color: #000000;
  font-weight: 700;
  font-size: 20px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 28px;
  margin-bottom: 8px;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c0 {
  margin-top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<h3
    class="c0 c1"
  >
    Foo
  </h3>,
  .c1 {
  margin: 0;
  color: #000000;
  font-weight: 700;
  font-size: 18px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 25px;
  margin-bottom: 8px;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c0 {
  margin-top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<h4
    class="c0 c1"
  >
    Foo
  </h4>,
]
`;
